src/NuGet.Core/NuGet.Commands/SignCommand/SignCommandRunner.cs

```diff
@@ -0,0 +1,130 @@
+
+            foreach (var packagePath in packagesToSign)
+            {
+                try
+                {
+                    SignPackage(packagePath, signArgs.Logger, signRequest);
+                }
+                catch(Exception e)
+                {
+                    success = false;
+                    ExceptionUtilities.LogException(e, signArgs.Logger);
+                }
+            }
+
+            if (success)
+            {
+                signArgs.Logger.LogInformation(Strings.SignCommandSuccess);
+            }
+            
+            return success ? 0 : 1;
+        }
+
+        private int SignPackage(string path, ILogger logger, SignPackageRequest request)
+        {
+            using (var zip = ZipFile.Open(path, ZipArchiveMode.Update))
+            {
+                var package = new SignedPackageArchive(zip);
+                var signer = new Signer(package);
+                signer.SignAsync(request, logger, CancellationToken.None).Wait();
+            }
+
+            return 0;
+        }
+
+        private SignPackageRequest GenerateSignPackageRequest(SignArgs signArgs, X509Certificate2 certificate)
+        {
+            return new SignPackageRequest()
+            {
+                Signature = new Signature()
+                {
+                    DisplayName = "FakeSignature",
+                    Type = SignatureType.Author,
+                    TestTrust = SignatureVerificationStatus.Trusted
+                }
+            };
+        }
+
+        private static X509Certificate2 GetCertificate(SignArgs signArgs)
+        {
+            var certFindOptions = new CertificateSourceOptions()
+            {
+                CertificatePath = signArgs.CertificatePath,
+                CertificatePassword = signArgs.CertificatePassword,
+                Fingerprint = signArgs.CertificateFingerprint,
+                StoreLocation = signArgs.CertificateStoreLocation,
+                StoreName = signArgs.CertificateStoreName,
+                SubjectName = signArgs.CertificateSubjectName
+            };
+
+            // get matching certificates
+            var matchingCertCollection = CertificateProvider.GetCertificates(certFindOptions);
+
+            if (matchingCertCollection.Count > 1)
+            {
+#if !IS_CORECLR
+                if (signArgs.NonInteractive || !RuntimeEnvironmentHelper.IsWindows)
+                {
+                    // if on non-windows os or in non interactive mode - display the certs and error out
+                    signArgs.Logger.LogInformation(CertificateUtility.X509Certificate2CollectionToString(matchingCertCollection));
+                    throw new InvalidOperationException(string.Format(Strings.SignCommandMultipleCertException, nameof(SignArgs.CertificateFingerprint)));
+                }
+                else
+                {
+                    // Else launch UI to select
+                    matchingCertCollection = X509Certificate2UI.SelectFromCollection(
```

Comment:
I dont think it will return multiple for `X509SelectionFlag.SingleSelection`. Ref - https://msdn.microsoft.com/en-us/library/system.security.cryptography.x509certificates.x509selectionflag(v=vs.110).aspx